parquet-hadoop/src/main/java/org/apache/parquet/hadoop/api/WriteSupport.java

```diff
@@ -121,6 +121,18 @@ public FinalizedWriteContext(Map<String, String> extraMetaData) {
   public abstract void write(T record);
 
   /**
+   * Called to get a name to identify the WriteSupport object model.
+   * If not null, this is added to the file footer metadata.
+   * <p>
+   * Defining this method will be required in a future API version.
+   *
+   * @return a String name for file metadata.
+   */
+  public String getName() {
+    return null;
+  }
```

Comment:
My reasoning is that using the constructor requires updates in a lot more places and it would be easier to overlook cases that need it.